boards/BMS/Src/App/states/App_AllStates.c

```diff
@@ -136,21 +140,27 @@ bool App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
         App_Airs_IsAirNegativeClosed(airs) ? CONTACTOR_STATE_CLOSED : CONTACTOR_STATE_OPEN);
     App_CanTx_BMS_Contactors_AirPositive_Set(
         App_Airs_IsAirPositiveClosed(airs) ? CONTACTOR_STATE_CLOSED : CONTACTOR_STATE_OPEN);
-    App_SetPeriodicCanSignals_Imd(imd);
 
+    App_SetPeriodicCanSignals_Imd(imd);
     App_AdvertisePackPower(accumulator, ts);
 
     App_CanTx_BMS_OkStatuses_BmsOk_Set(App_OkStatus_IsEnabled(bms_ok));
     App_CanTx_BMS_OkStatuses_ImdOk_Set(App_OkStatus_IsEnabled(imd_ok));
     App_CanTx_BMS_OkStatuses_BspdOk_Set(App_OkStatus_IsEnabled(bspd_ok));
 
+    const bool dcm_fault      = App_CanAlerts_BoardHasFault(DCM_ALERT_BOARD);
+    const bool fsm_fault      = App_CanAlerts_BoardHasFault(FSM_ALERT_BOARD);
+    const bool pdm_fault      = App_CanAlerts_BoardHasFault(PDM_ALERT_BOARD);
+    const bool dim_fault      = App_CanAlerts_BoardHasFault(DIM_ALERT_BOARD);
+    const bool fault_over_can = dcm_fault || fsm_fault || pdm_fault || dim_fault;
```

Comment:
isn't the heartbeat also over CAN? I see what you're trying to say here but I think something along the lines of "fault_from_other_board" would work better (ik that's a bit long, open to something more concise)

Reply:
yeah good call ill change it